Extended guild classes

cloudrac3r opened the pull request:
Closes #215. This doesn't address the comment about guild names.

If you want to optimise out the ClassList thing into string matching functions then go ahead, but JS is fast so I don't care enough to do it myself.

The render function is called on a guild (and only that guild) when it is hovered or selected, which was very convenient for the creation of this plugin.

cloudrac3r committed:
Ignore vscode dotfiles

cloudrac3r committed:
Create pc-guild-classes plugin

cloudrac3r commented:
fuck you eslint i do what i want

cloudrac3r committed:
how 2 lint

cyyynthia commented on src/Powercord/plugins/pc-guild-classes/index.js:
```diff
@@ -1,57 +1,56 @@
+      if (res._owner) {
+        const { hovered } = res._owner.memoizedState;
+        const { selected } = res._owner.pendingProps;
+        const classList = new ClassList(res.props.className);
```
is an extra class required? Wouldn't something like this enough?
```js
const classNames = [ ...new Set(className.split(' ')) ]
hovered && classNames.push('...')
selected && classNames.push('...')
res.props.className = classNames.join(' ');
```

cyyynthia commented on src/Powercord/plugins/pc-guild-classes/index.js:
```diff
@@ -1,57 +1,56 @@
+  }
+
+  async _patchGuild () {
+    const guildInner = await waitFor('.blobContainer-239gwq');
```
Avoid using raw classes in here, fetch them dynamically

cyyynthia commented on src/Powercord/plugins/pc-guild-classes/index.js:
```diff
@@ -1,57 +1,56 @@
+    const guildInner = await waitFor('.blobContainer-239gwq');
+    const guildElement = guildInner.parentElement;
+    const Guild = getOwnerInstance(guildElement);
+    inject('cadence-guild-classes', Guild.constructor.prototype, 'render', (_, res) => {
```
Keeping injection names in here with "pc-" would be more consistent as it's an internal plugin

cyyynthia commented on src/Powercord/plugins/pc-guild-classes/index.js:
```diff
@@ -1,57 +1,56 @@
+        const { hovered } = res._owner.memoizedState;
+        const { selected } = res._owner.pendingProps;
+        const classList = new ClassList(res.props.className);
+        classList.conditional(hovered, 'pgc-hovered');
```
is a prefix really required here? wouldn't just go for .hovered be enough?

cloudrac3r committed:
Changes as requested
- Remove class name builder
- Use pc- for injection
- Get blobContainer class from module
- Drop pgc- prefix from classes

GriefMoDz committed:
Relocate guild classes to 'pc-classNameNormalizer'

* Remove 'pc-guild-classes' and relocate past functions to 'pc-classNameNormalizer'.
* Add guild and folder labels for extended CSS support.
